### PR TITLE
Add support for the latest version OracleManagedDataAccess

### PR DESCRIPTION
**Pull request by z1c0**
Use latest version for `Oracle.ManagedDataAccess.Core`

### CODE REVIEW

File: sample/OracleManagedDataAccessCoreSample/OracleManagedDataAccessCoreSample.csproj [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <OracleManagedDataAccessCoreVersion Condition="'$(OracleManagedDataAccessCoreVersion)'==''">2.19.120</OracleManagedDataAccessCoreVersion>
+    <OracleManagedDataAccessCoreVersion Condition="'$(OracleManagedDataAccessCoreVersion)'==''">3.21.90</OracleManagedDataAccessCoreVersion>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>net461;netcoreapp3.1;net5.0;</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
```
Comment (z1c0) [inline sample/OracleManagedDataAccessCoreSample/OracleManagedDataAccessCoreSample.csproj:6]:
```suggestion
    <TargetFrameworks>net461;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>

```

File: sample/OracleManagedDataAccessCoreSample/OracleManagedDataAccessCoreSample.csproj [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <OracleManagedDataAccessCoreVersion Condition="'$(OracleManagedDataAccessCoreVersion)'==''">2.19.120</OracleManagedDataAccessCoreVersion>
+    <OracleManagedDataAccessCoreVersion Condition="'$(OracleManagedDataAccessCoreVersion)'==''">3.21.90</OracleManagedDataAccessCoreVersion>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>net461;netcoreapp3.1;net5.0;</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
```
Comment (z1c0) [inline sample/OracleManagedDataAccessCoreSample/OracleManagedDataAccessCoreSample.csproj:6]:
```suggestion
    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>

```

**Comment by z1c0** [comment]
CI is passing finally 🥳 .

**Review by bmorelli25 — APPROVED** [review]
Doc changes LG

**Review by gregkalapos — APPROVED** [review]
LGTM.

Is the solution for fixing the CI issues to drop `net461` in https://github.com/elastic/apm-agent-dotnet/pull/1968/commits/53f011885c36d8d255d7cb55da219ee7b783b7ce? If I remember correctly it only failed on Linux, so those should have not run there, right?

**Pull request merged**